core/templates/pages/blog-dashboard-page/blog-dashboard-page.component.spec.ts

```diff
@@ -239,4 +255,99 @@ describe('Blog Dashboard Page Component', () => {
         'Unable to create new blog post.Error: ' +
         'To many collisions with existing blog post ids.');
     }));
+
+  it('should remove unpublish blog post from published list and' +
+  ' add it to drafts list', () => {
+    let summaryObject = BlogPostSummary.createFromBackendDict(
+      { id: 'sampleId',
+        author_username: 'test_user',
+        title: 'Title',
+        summary: 'Hello World',
+        tags: ['news'],
+        thumbnail_filename: 'image.png',
+        url_fragment: 'title',
+        last_updated: '3232323',
+        published_on: '3232323',
+      });
+    let blogDashboardData = {
+      username: 'test_user',
+      profilePictureDataUrl: 'sample_url',
+      numOfPublishedBlogPosts: 0,
+      numOfDraftBlogPosts: 0,
+      publishedBlogPostSummaryDicts: [],
+      draftBlogPostSummaryDicts: [],
+    };
+    component.blogDashboardData = blogDashboardData;
+    component.blogDashboardData.publishedBlogPostSummaryDicts = [summaryObject];
+    component.blogDashboardData.draftBlogPostSummaryDicts = [];
+    component.blogDashboardData.numOfDraftBlogPosts = 0;
+    component.blogDashboardData.numOfPublishedBlogPosts = 1;
+
+    component.unpublishedBlogPost(summaryObject);
+
+    expect(component.blogDashboardData.draftBlogPostSummaryDicts)
+      .toEqual([summaryObject]);
+    expect(component.blogDashboardData.publishedBlogPostSummaryDicts)
+      .toEqual([]);
```

Comment:
This value doesn't seem to change. If this is expected please place a comment explaining why in the code

Reply:
Refactored the code a bit for easy understanding and also added comment PTAL.